### PR TITLE
[General] Can I add Machine Learning Algorithms? 

### PR DESCRIPTION
**Pull request by inkerton**
### Description
The Decision Tree Algorithm is a simple yet powerful model that can be used for both classification and regression tasks. Its ease of interpretation, ability to handle both categorical and numerical data, and non-parametric nature make it a versatile choice for many machine learning problems. However, decision trees are prone to overfitting, which can be mitigated by techniques like pruning, random forests, or ensemble methods.

Fixes # (issue number) #643 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: MachineLearning/decisionTree.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
+id: decision-tree
```
Comment (ajay-dhangar) [inline MachineLearning/decisionTree.md:2]: write before line `1` ``` --- ``` for our docs format

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: MachineLearning/decisionTree.md (hunk 1 above)
Comment (inkerton) [reply MachineLearning/decisionTree.md:2]: Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**